dot/state/service.go

```diff
@@ -112,23 +132,22 @@ func initializeBlockTree(db database.Database, genesisHeader *types.Header) erro
 
 // Start initializes the Storage database and the Block database.
 func (s *Service) Start() error {
-	if s.Storage != nil || s.Block != nil || s.Network != nil {
-		return nil
-	}
+	db := s.db
+	if !s.isMemDB {
```

Comment:
is this being used for testing purposes only at the moment? should we make note in a comment here?

Reply:
yes, it's only for testing purposes. I will make a note